network/manager.go

```diff
@@ -771,6 +772,21 @@ func (nm *networkManager) DeleteState(_ []*EndpointInfo) error {
 	// For stateless cni, plugin.ipamInvoker.Delete takes care of removing the state in the main Delete function
 
 	if nm.IsStatelessCNIMode() {
+		for _, epInfo := range epInfos {
+			// this cleanup happens only for standalone swiftv2 to delete endpoint state from CNS.
+			if epInfo.NICType == cns.NodeNetworkInterfaceFrontendNIC || epInfo.NICType == cns.NodeNetworkInterfaceAccelnetFrontendNIC {
+				response, err := nm.CnsClient.DeleteEndpointState(context.TODO(), epInfo.EndpointID)
+				if err != nil {
+					if response != nil && response.ReturnCode == types.NotFound {
+						logger.Info("Endpoint state not found in CNS", zap.String("endpointID", epInfo.EndpointID))
+						return nil
+					}
+					return errors.Wrapf(err, "Delete endpoint API returned with error for endpoint %s", epInfo.EndpointID)
```

Comment:
This error here will fail the delete call, so for whatever reason (CNS is down for example) if the endpoint cannot get deleted form statefile the whole Delete will fall in a loop. We have to be careful about catching errors from CNS here.